cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaMirrorMaker2Connectors.java

```diff
@@ -326,4 +330,18 @@ private static String addTLSConfigToMirrorMaker2ConnectorConfig(Map<String, Obje
             return "PLAINTEXT";
         }
     }
+
+    private static boolean hasMatchingBootstrapServers(KafkaMirrorMaker2 kafkaMirrorMaker2, String connectClusterAlias, String targetClusterAlias) {
+        List<String> configs = Optional.ofNullable(kafkaMirrorMaker2)
+                .map(KafkaMirrorMaker2::getSpec)
+                .map(KafkaMirrorMaker2Spec::getClusters)
+                .orElse(Collections.emptyList())
+                .stream()
+                .filter(cluster -> connectClusterAlias.equals(cluster.getAlias()) || targetClusterAlias.equals(cluster.getAlias()))
+                .map(KafkaMirrorMaker2ClusterSpec::getBootstrapServers)
+                .toList();
+
+        return configs.size() == 2 && configs.get(0).equals(configs.get(1));
```

Comment:
This looks super complex and hard to read. Why don't you pass the list of clusters here, find the one for the first alias, find the one for the second alias and then compare them?